lib/ui/survey_question/survey_questions_screen.dart

```diff
@@ -103,10 +133,22 @@ class _SurveyQuestionsState extends ConsumerState<SurveyQuestionsScreen> {
       alignment: Alignment.bottomRight,
       child: CircleNextButton(
         onPressed: () {
-          // TODO: Show the next question
-          showToastMessage("Next question");
+          ref
+              .read(surveyQuestionsViewModelProvider.notifier)
+              .getNextSurveyQuestion(_currentIndex + 1);
+          _currentIndex++;
```

Comment:
How about creating a stream for the `_currentIndex` and managing it inside the ViewModel? 🤔 The `_currentIndex` and `surveyQuestion` might not match each other somehow 💭 
And I think for the `getNextSurveyQuestion`, the next should be always `current + 1` so we don't need to pass the index, otherwise it should be `getSurveyQuestionAt(index)`

Reply:
> How about creating a stream for the _currentIndex and managing it inside the ViewModel?

@hoangnguyen92dn, I have thought about this way too, so as you asked, I will create stream `_currentIndex` instead of keeping the model. 🚀